### PR TITLE
Add home page functionality

### PR DESCRIPTION
**Pull request by matthew-sheehan**

### CODE REVIEW

File: src/utils/getTopics.js [hunk 1]
```diff
@@ -0,0 +1,8 @@
+import axios from 'axios'
+
+const getTopics = () => {
```
Comment (NRMSMITH) [inline src/utils/getTopics.js]: I would suggest that this doesn't need to be in a different file - you could just have a single apicalls file :)

File: src/components/Home.jsx [hunk 2]
```diff
@@ -0,0 +1,43 @@
+import { useState, useEffect } from 'react'
+import SearchBar from './SearchBar'
+import { getArticles } from '../utils/getArticles'
+import ArticlePreview from './ArticlePreview'
+
+const Home = ({ topics }) => {
+    const [searchResults, setSearchResults] = useState([])
+    const [resultsPage, setResultsPage] = useState(1)
+    const [totalPages, setTotalPages] = useState(0)
+
```
Comment (NRMSMITH) [inline src/components/Home.jsx:12]: an isLoading state would be really beneficial here

**Review by NRMSMITH — CHANGES_REQUESTED** [review]
Your design is really responsive which is lovely - what I would suggest is that you need to have a loading state created. I would also suggest having your apicalls in one file :) Please submit another PR when you have completed this tomorrow morning :) (not during the lecture!!)

File: src/utils/apis.js [hunk 3]
```diff
@@ -0,0 +1,11 @@
+import axios from 'axios'
+
+const api = axios.create({ baseURL: 'https://nc-news-api-45fk.onrender.com/api'})
+
+export const getArticles = (resultsPage) => {
+    return api.get(`/articles?p=${resultsPage}&total_count=true`).then(res => res.data)
```
Comment (itskatherine) [inline src/utils/apis.js:6]: Nice that you've split out this functionality into an api file ⭐  - you might want to consider using the [axios params object](https://notes.northcoders.com/courses/js-front-end/axios#axios-params), to save using a template literal here, up to you though!

File: src/components/SearchBar.jsx [hunk 4]
```diff
@@ -0,0 +1,15 @@
...
+        <form id="searchbar">
+            <label htmlFor="topic-filter-dropdown">Filter by Topic: </label>
+            <select name="topic" id="topic-filter-dropdown">
+                {topics.map(topic => {
+                    return <option key={topic.slug} value={topic.slug}>{topic.slug}</option>
+                })}
+            </select>
+            <button>Search</button>
+        </form>
+    </section>
```
Comment (itskatherine) [inline src/components/SearchBar.jsx:12]: Though this is good to add, I would say that this element (though it isn't working yet) isn't in the scope of this ticket/task (it is in the scope of some tickets further on) - and it's important in PRs for the code just to meet the requirements of that ticket. Going forward try to keep this in mind 🧠

**Review by itskatherine — COMMENTED** [review]
Love the pagination Matt - a few comments to look over but once you have you're good to merge 👍

**Pull request merged**